src/libraries/System.Security.Cryptography.Xml/src/System/Security/Cryptography/Xml/CanonicalXmlDocument.cs

```diff
@@ -61,52 +61,54 @@ public void WriteHash(HashAlgorithm hash, DocPosition docPos, AncestralNamespace
             }
         }
 
+#pragma warning disable CS8765 // Nullability doesn't match overriden member
         public override XmlElement CreateElement(string prefix, string localName, string namespaceURI)
+#pragma warning restore CS8765
         {
             return new CanonicalXmlElement(prefix, localName, namespaceURI, this, _defaultNodeSetInclusionState);
         }
 
-        public override XmlAttribute CreateAttribute(string prefix, string localName, string namespaceURI)
+        public override XmlAttribute CreateAttribute(string? prefix, string localName, string? namespaceURI)
         {
             return new CanonicalXmlAttribute(prefix, localName, namespaceURI, this, _defaultNodeSetInclusionState);
         }
 
-        protected override XmlAttribute CreateDefaultAttribute(string prefix, string localName, string namespaceURI)
+        protected override XmlAttribute CreateDefaultAttribute(string? prefix, string localName, string? namespaceURI)
         {
             return new CanonicalXmlAttribute(prefix, localName, namespaceURI, this, _defaultNodeSetInclusionState);
         }
 
-        public override XmlText CreateTextNode(string text)
+        public override XmlText CreateTextNode(string? text)
         {
             return new CanonicalXmlText(text, this, _defaultNodeSetInclusionState);
         }
 
-        public override XmlWhitespace CreateWhitespace(string prefix)
+        public override XmlWhitespace CreateWhitespace(string? prefix)
         {
-            return new CanonicalXmlWhitespace(prefix, this, _defaultNodeSetInclusionState);
+            return new CanonicalXmlWhitespace(prefix!, this, _defaultNodeSetInclusionState);
         }
 
-        public override XmlSignificantWhitespace CreateSignificantWhitespace(string text)
+        public override XmlSignificantWhitespace CreateSignificantWhitespace(string? text)
         {
-            return new CanonicalXmlSignificantWhitespace(text, this, _defaultNodeSetInclusionState);
+            return new CanonicalXmlSignificantWhitespace(text!, this, _defaultNodeSetInclusionState);
         }
 
         public override XmlProcessingInstruction CreateProcessingInstruction(string target, string data)
         {
             return new CanonicalXmlProcessingInstruction(target, data, this, _defaultNodeSetInclusionState);
         }
 
-        public override XmlComment CreateComment(string data)
+        public override XmlComment CreateComment(string? data)
         {
-            return new CanonicalXmlComment(data, this, _defaultNodeSetInclusionState, _includeComments);
+            return new CanonicalXmlComment(data!, this, _defaultNodeSetInclusionState, _includeComments);
```

Comment:
Why is this being `!`'d?  CanonicalXmlComment's ctor should have a `string? comment` parameter.